### PR TITLE
Fix RuntimeWarning: coroutine 'OctopusIntelligentSystem.start' was never awaited

### PR DESCRIPTION
**Pull request by pdcastro**
Issue #28 had already been closed as [“hopefully fixed”](https://github.com/megakid/ha_octopus_intelligent/issues/28#issuecomment-1925498437), but it remained. This PR fixes that issue and another issue that got uncovered by the fix.

### CODE REVIEW

File: custom_components/octopus_intelligent/__init__.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     )
 
     try:
-        await hass.async_add_executor_job(octopus_system.start)
+        await octopus_system.start()
```
Comment (pdcastro) [inline custom_components/octopus_intelligent/__init__.py:49]: This is the fix for HASS log messages such as:
```
2024-02-25 15:05:06.334 WARNING (MainThread) [py.warnings] /usr/local/lib/python3.12/asyncio/base_events.py:1959: RuntimeWarning: coroutine 'OctopusIntelligentSystem.start' was never awaited
  handle = self._ready.popleft()
```
Or:
```
2024-02-15 19:03:02.926 WARNING (MainThread) [py.warnings] /usr/local/lib/python3.12/asyncio/base_events.py:1975: RuntimeWarning: coroutine 'OctopusIntelligentSystem.start' was never awaited
  handle = None  # Needed to break cycles when an exception occurs.
```
The `start()` method was simply not being called. `async_add_executor_job()` creates a new thread to run the given function, but it does not expect the function to be asynchronous and does not `await` it. When not awaited, asynchronous functions like `start()` are not executed.

File: custom_components/octopus_intelligent/__init__.py [hunk 2]
```diff
@@ -60,9 +60,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     #hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, lambda event: octopus_system.stop())
 
-    for component in PLATFORMS:
-        hass.async_create_task(
-            hass.config_entries.async_forward_entry_setup(entry, component))
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
```
Comment (pdcastro) [inline custom_components/octopus_intelligent/__init__.py:63]: This is not a fix for anything, just an incidental simplification as `async_forward_entry_setups()` includes the `for` loop: https://github.com/home-assistant/core/blob/2024.2.5/homeassistant/config_entries.py#L1566

File: custom_components/octopus_intelligent/graphql_client.py [hunk 3]
```diff
@@ -191,7 +191,7 @@ async def __async_cancel_boost_charge(self, session, account_id: str):
     result = await session.execute(query, variable_values=params, operation_name="deleteBoostCharge")
     return result['deleteBoostCharge']
 
-  async def __async_get_accounts(session):
+  async def __async_get_accounts(self, session):
```
Comment (pdcastro) [inline custom_components/octopus_intelligent/graphql_client.py:194]: The `start()` function was not being executed, as it wasn’t awaited. When `start()` is awaited (the prominent fix in this PR), it executes other code and reveals other issues, like this issue (`__async_get_accounts() takes 1 positional argument but 2 were given`):
```
2024-02-25 23:25:02.662 ERROR (MainThread) [custom_components.octopus_intelligent.octopus_intelligent_system] Authentication failed : OctopusEnergyGraphQLClient.__async_get_accounts() takes 1 positional argument but 2 were given. You may need to check your token or create a new app in the gardena api and use the new token.
```

File: custom_components/octopus_intelligent/octopus_intelligent_system.py [hunk 4]
```diff
@@ -179,7 +179,7 @@ async def start(self):
                 _LOGGER.error(f"Account {self._account_id} not found in accounts {accounts}")
                 raise Exception(f"Account {self._account_id} not found in accounts {accounts}")
         except Exception as ex:
-            _LOGGER.error(f"Authentication failed : {ex.message}. You may need to check your token or create a new app in the gardena api and use the new token.")
+            _LOGGER.error(f"Authentication failed : {ex}. You may need to check your token or create a new app in the gardena api and use the new token.")
```
Comment (pdcastro) [inline custom_components/octopus_intelligent/octopus_intelligent_system.py:182]: Similarly as the other comment, when `start()` is executed it revealed other issues. Here the issue was `'TypeError' object has no attribute ‘message’`:
```
2024-02-25 23:22:42.043 ERROR (MainThread) [custom_components.octopus_intelligent] Got error when setting up Octopus Intelligent Integration: 'TypeError' object has no attribute ‘message'
```

This log message and most of the `start()` function is further modified / improved in another PR: #40

**Review by megakid — APPROVED** [review]

**Pull request merged**